File: app/src/main/java/com/josh/billrssroom/ui/FavoritesAdapter.java
package com.josh.billrssroom.ui;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.josh.billrssroom.R;
import com.josh.billrssroom.databinding.ItemRowFavoritesBinding;
import com.josh.billrssroom.model.BillItem;

import java.util.List;

public class FavoritesAdapter extends RecyclerView.Adapter<FavoritesViewHolder> {

    private List<BillItem> billEntities;

    @Nullable
    private final FavoriteClickCallback favoriteClickCallback;

    public FavoritesAdapter(Context context, @Nullable FavoriteClickCallback favoriteClickCallback) {
        this.favoriteClickCallback = favoriteClickCallback;
    }

    @NonNull
    @Override
    public FavoritesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        ItemRowFavoritesBinding binding = DataBindingUtil
                .inflate(LayoutInflater.from(parent.getContext()), R.layout.item_row_favorites,
                        parent, false);
        binding.setCallback(favoriteClickCallback);
        return new FavoritesViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull FavoritesViewHolder holder, int position) {
        BillItem item = billEntities.get(position);
        holder.binding.setBill(billEntities.get(position));
        holder.binding.executePendingBindings();
    }

    public void setBillEntities(List<BillItem> billEntities) {
        this.billEntities = billEntities;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        if (billEntities != null)
            return billEntities.size();
        else return 0;
    }
}

File: app/src/main/java/com/josh/billrssroom/screens/favorites/FavoritesAdapter.kt
package com.josh.billrssroom.screens.favorites

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.josh.billrssroom.R
import com.josh.billrssroom.model.FeedItem
import kotlinx.android.synthetic.main.include_item_row_fav.view.*
import java.util.*

class FavoritesAdapter(private val activity: Activity,
                       val favoriteClickCallback: FavoriteClickListener) :
    RecyclerView.Adapter<FavoritesAdapter.FavoritesViewHolder>() {

    private var favoritesList: List<FeedItem> = Collections.emptyList()
    private var expandedPosition = -1


    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): FavoritesViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        return FavoritesViewHolder(layoutInflater.inflate(R.layout.item_row_favorites, parent, false))
    }

    override fun onBindViewHolder(holder: FavoritesViewHolder, position: Int) {
        val items = favoritesList[position]

        holder.bindTo(items)

//        if (position == expandedPosition){
//            holder.groupButtons.visibility = View.VISIBLE
//        } else{
//            holder.groupButtons.visibility = View.GONE
//        }
    }

    fun setFavoritesList(favoriteItemsList: List<FeedItem>) {
        favoritesList = favoriteItemsList
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return favoritesList.size
    }

    inner class FavoritesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val titleView = itemView.text_title
        val dateView = itemView.text_date
        val descriptionView = itemView.text_description
        val btnShare = itemView.btn_share
        val btnBrowser = itemView.btn_browser
        val btnTrash = itemView.btn_trash
        val groupButtons = itemView.group
        var favorite: FeedItem? = null

        fun bindTo(favorite: FeedItem){
            this.favorite = favorite
            titleView.text = favorite.title
            dateView.text = favorite.formattedDate
            descriptionView.text = favorite.formattedDescription

            btnTrash.setOnClickListener { favoriteClickCallback.onTrashBtnClick(favorite, adapterPosition) }
            btnBrowser.setOnClickListener { favoriteClickCallback.onBrowserBtnClick(favorite, adapterPosition) }
            btnShare.setOnClickListener { favoriteClickCallback.onShareBtnClick(favorite, adapterPosition) }

//            itemView.setOnClickListener {
//                if (expandedPosition >= 0){
//                    val prev = expandedPosition
//                    notifyItemChanged(prev)
//                }
//                expandedPosition = adapterPosition
//                notifyItemChanged(expandedPosition)
//            }
        }
    }
}